data/systemd/snap.mount.in

```diff
@@ -0,0 +1,14 @@
+# ensure @SNAP_MOUNT_DIR@ MS_SHARED so that lxd works correctly (LP: #1668659)
+[Unit]
+Description=Ensure that the snap directory shares mount events.
+ConditionVirtualization=lxc
```

Comment:
Should we use "container" here instead? Sounds like a good bet, as I don't think lxd isn't doing much more than  having standard initialization of the stock scripts inside a container.

If so, let's also adjust the comment above to say container instead of "lxd".

Reply:
I didn't test this in any other container. It depends on how those start up stuff. For now I switched it to `container` but I think we ought to do at least some small peek at, e.g. docker.

Reply:
Docker doesn't go through the normal startup logic. They'd not be affected by this either way, I believe.